### PR TITLE
canvas integration teacher and admin states

### PR DESCRIPTION
**Pull request by emilia-friedberg**
## WHAT
Implement new teacher and admin states for the Canvas integration

## WHY
To give teachers and admins as much information as possible about how to set up a Canvas integration, if they'd like to.

## HOW
Update both My Classes page and the import students section on the Assign Students page to give users the option of all three integrations to import from if they are unaffiliated. If they select Canvas, open a modal with a variety of states that address each potential next step for a user to complete if they wish to use Canvas.

### Screenshots
When the teacher does not have school or district premium (A/A1 from spec)
canvas_a

When the teacher's school is not linked to Canvas (B from spec)
canvas_b

When the teacher is not personally linked to Canvas (C from spec)
canvas_c

When the admin does not have school or district premium (E/E1 from spec)
canvas_e

When the admin's school is not linked to Canvas (F from spec)
canvas_f

When the admin is not individually linked to Canvas (G from spec)
canvas_g

### Notion Card Links
https://www.notion.so/quill/Implement-new-teacher-and-admin-states-for-the-Canvas-integration-cd214f6450474aa09272c966475e6c21?pvs=4

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | YES

### CODE REVIEW

File: services/QuillLMS/client/app/bundles/Teacher/components/classrooms/__tests__/canvas_modal.test.jsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import CanvasModal from '../canvas_modal'; // Adjust the import path as necessary
+
+const TEACHER = 'teacher'
+const ADMIN = 'admin'
+const TEACHER_PREMIUM = 'Teacher Premium'
+const SCHOOL_PREMIUM = 'School Premium'
```
Comment (eadams17) [inline services/QuillLMS/client/app/bundles/Teacher/components/classrooms/__tests__/canvas_modal.test.jsx:9]: I think we already have these consts defined elsewhere

File: services/QuillLMS/client/app/bundles/Teacher/components/classrooms/canvas_modal.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+          <div className="right-aligned-buttons">
+            {canvasStates[canvasState].buttons}
+          </div>
+        </div>
+      </div>
+    </div>
+  )
+}
+
+export default CanvasModal;
```
Comment (eadams17) [inline services/QuillLMS/client/app/bundles/Teacher/components/classrooms/canvas_modal.tsx:176]: love the way how all of this is organized!

**Review by eadams17 — APPROVED** [review]
Looks good to me!

**Review by anathomical — APPROVED** [review]
Nice job organizing the logic for a pretty complex interaction of states.

File: services/QuillLMS/client/app/bundles/Teacher/components/classrooms/canvas_modal.tsx (hunk 2 above)
Comment (brendanshean) [reply services/QuillLMS/client/app/bundles/Teacher/components/classrooms/canvas_modal.tsx:176]: Agreed.  Super easy to follow.

File: services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx [hunk 3]
```diff
@@ -158,7 +163,7 @@ const ActiveClassrooms = ({
       setPendingImportFromProviderRequest(true)
       retrieveProviderClassrooms()
     } else {
-      openModal(linkCanvasAccountModal)
```
Comment (brendanshean) [inline services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx:161]: Thanks for fixing my bug there.

File: services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx [hunk 4]
```diff
@@ -407,19 +412,14 @@ const ActiveClassrooms = ({
...
-          {renderImportFromProviderButton(googleProvider)}
+          {provider ? renderImportFromProviderButton(provider) : renderImportClassesDropdown()}
           {renderCreateAClassButton()}
         </div>
       </div>
     )
   }
 
   const renderImportFromProviderButton = (theProvider: string) => {
-    if (provider && provider != theProvider) { return null }
```
Comment (brendanshean) [inline services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx:420]: Nice work on cleaning this up.

**Review by brendanshean — APPROVED** [review]
Great attention to detail managing all of these complicated states.

**Pull request merged**